### PR TITLE
chore: remove deprecated "io/ioutil" imports

### LINKED ISSUE #25059: Remove deprecated "io/ioutil" imports
`"io/ioutil"` is deprecated since go1.6. This ticket is about going through the codebase and migrating all imports to the new packages like `os`.

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-55067

### PR DESCRIPTION
**Pull request by Dhoni77**
#### Summary

Package `ioutil` is deprecated as of 1.16.
So migrated all imports in the application to `os` package

Below are the methods or properties which have changed:

| io/ioutil | os         |
|-----------|------------|
| TempDir   | MkdirTemp  |
| ReadFile  | ReadFile   |
| WriteFile | WriteFile  |
| Discard   | Discard    |
| TempFile  | CreateTemp |

#### Ticket Link

Fixes #25059

#### Release Note

Deprecation 

```release-note
NONE
```

### CODE REVIEW

**Comment by hanzei** [comment]
@Dhoni77 Please let me know when the PR is ready for review

**Comment by Dhoni77** [comment]
Hi @hanzei the pr is ready for review.

File: server/cmd/mmctl/commands/sampledata.go [hunk 1]
```diff
@@ -179,8 +182,12 @@ func processProfileImagesDir(profileImagesPath, tmpDir, bulk string) ([]string,
 		// we're not importing the resulting file, so we keep the
 		// image paths corresponding to the value of the flag
 	} else {
-		for _, profileImage := range profileImagesFiles {
-			profileImages = append(profileImages, filepath.Join(profileImagesPath, profileImage.Name()))
+		for _, dirEntry := range dirEntries {
+			fileInfo, err := dirEntry.Info()
```
Comment (hanzei) [inline server/cmd/mmctl/commands/sampledata.go:186]: Why is `fs.FileInfo` needed here? Is `os.DirEntry` enough?

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the contribution +1: 

One question regarding the migration of `ioutil.ReadDir` below

File: server/cmd/mmctl/commands/sampledata.go (hunk 1 above)
Comment (Dhoni77) [reply server/cmd/mmctl/commands/sampledata.go:186]: Previously ioutil.ReadDir was returning FileInfo[] but os.ReadDir returns DirEntry[] so I wanted to keep the existing return type without breaking anything.
https://pkg.go.dev/io/ioutil#ReadDir
https://pkg.go.dev/os#ReadDir

Comment (hanzei) [reply server/cmd/mmctl/commands/sampledata.go:186]: Would you please check if `DirEntry[]` is enough here? The code will fail to compile if any field is missing.

Comment (Dhoni77) [reply server/cmd/mmctl/commands/sampledata.go:186]: The build was successful when i ran locally.
In the docs also return type is mentioned as []DirEntry and pipelines are also successful👍

Comment (hanzei) [reply server/cmd/mmctl/commands/sampledata.go:186]: Awesome to hear that `DirEntry` is enough here. Could you also check the other code pieces where you migrated `ioutil.ReadDir` and check if `DirEntry` is enough there?

**Comment by hanzei** [comment]
@Dhoni77 Please let me know if you have any questions regarding my feedback.

**Comment by hanzei** [comment]
@Dhoni77 Heads up that I implemented the request changes from https://github.com/mattermost/mattermost/pull/25119#discussion_r1372938991 in https://github.com/mattermost/mattermost/pull/25119/commits/d274e3592994f30196cfd941238d878b6c36c37f.

**Review by hanzei — APPROVED** [review]

**Review by isacikgoz — APPROVED** [review]
LGTM, thanks @Dhoni77 🎉

**Comment by hanzei** [comment]
Thanks for the contribution @Dhoni77 :+1:

**Pull request merged**